cub/detail/exec_check_disable.cuh

```diff
@@ -0,0 +1,41 @@
+/*
+*  Copyright 2021 NVIDIA Corporation
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*/
+
+#pragma once
+
+#include <cub/util_compiler.cuh>
+
+/**
+ * \def CUB_EXEC_CHECK_DISABLE
+ * Wrapper around `#pragma nv_exec_check_disable`.
+ */
+
+// #pragma nv_exec_check_disable is only recognized by NVCC.
+#if defined(__CUDACC__) && \
+    !defined(_NVHPC_CUDA) && \
+    !(defined(__CUDA__) && defined(__clang__))
+
+#if CUB_HOST_COMPILER == CUB_HOST_COMPILER_MSVC
+#define CUB_EXEC_CHECK_DISABLE __pragma("nv_exec_check_disable")
```

Comment:
There's a bug 3423950 related to `__pragma` usage in nvcc 11.5 with `nv_diagnostic`. It leads to an internal compilation error on windows. I'm not sure if it's `nv_diagnostic` specific or not. Can you check this, please? If this issue is related to `__pragma` in general, we'll have to add another condition in the compiler check.

Reply:
We've tested `__pragma("nv_exec_check_disable")` in Thrust for the last several releases (including 11.5), it shouldn't have the same issue as 3423950.